pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
I had added it there in the first place to avoid scaling down the new replica set in case another controller with pods is adopted. It is the wrong place to handle it and the rollout code should take care of scaling down those RSs.